storage/innobase/buf/buf0flu.cc

```diff
@@ -1157,17 +1117,17 @@ static ulint buf_flush_try_neighbors(fil_space_t *space,
         ut_ad(!buf_pool.watch_is_sentinel(*b));
         ut_ad(b->oldest_modification() > 1);
       flush:
-        if (b->flush(evict, space))
+        if (b->flush(space))
         {
           ++count;
           continue;
         }
       }
       /* We avoid flushing 'non-old' blocks in an eviction flush,
       because the flushed blocks are soon freed */
-      else if ((!evict || b->is_old()) && !buf_pool.watch_is_sentinel(*b) &&
-               b->oldest_modification() > 1 && b->lock.u_lock_try(true))
+      else if (b->oldest_modification() > 1 && b->lock.u_lock_try(true))
       {
+        ut_ad(!buf_pool.watch_is_sentinel(*b));
```

Comment:
It would have been easier to understand the question if you had quoted multiple lines of source code:
```diff
-      else if ((!evict || b->is_old()) && !buf_pool.watch_is_sentinel(*b) &&
-               b->oldest_modification() > 1 && b->lock.u_lock_try(true))
+      else if (b->oldest_modification() > 1 && b->lock.u_lock_try(true))
       {
+        ut_ad(!buf_pool.watch_is_sentinel(*b));
```
For the dummy blocks in `buf_pool.watch[]`, we never modify `oldest_modification_`, so it is always 0 and the condition `b->oldest_modification() > 1` would not hold on them. Sure, I could add a comment about that:
```c++
        /* buf_pool.watch[i].oldest_modification() == 0 */
        ut_ad(!buf_pool.watch_is_sentinel(*b));
```

Reply:
Addressed in d2e8878faa5808cfacb019070868c3fd5db1a3d1. There is a redundant condition in `buf_flush_check_neighbor()` as well.